sled-agent/src/rack_setup/service.rs

```diff
@@ -1234,4 +1259,78 @@ impl ServiceInner {
 
         Ok(())
     }
+
+    // Query MGS servers in each switch zone to determine which switch slot they are managing.
+    // This logic does not handle an event where there are multiple racks. Is that ok?
+    async fn map_switch_zone_addrs(
+        &self,
+        switch_zone_addresses: Vec<Ipv6Addr>,
+    ) -> HashMap<SwitchLocation, Ipv6Addr> {
+        info!(self.log, "Determining switch slots managed by switch zones");
+        let mut switch_zone_addrs = HashMap::new();
+        for addr in switch_zone_addresses {
+            let mgs_client = MgsClient::new(
+                &format!("http://[{}]:{}", addr, MGS_PORT),
+                self.log.new(o!("component" => "MgsClient")),
+            );
+
+            info!(self.log, "determining switch slot managed by dendrite zone"; "zone_address" => format!("{addr:#?}"));
+            let switch_slot = loop {
```

Comment:
Neat, the thought of a `let thing = loop { ... break thing; ... }` type syntax for retry loops had not crossed my mind.

Reply:
Yeah, I was pleasantly surprised!